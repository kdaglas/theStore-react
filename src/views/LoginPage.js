import React, { Component } from "react";
import { connect } from "react-redux";
import Login from "../components/loginForm";
import loginAction from "../actions/loginAction";

export class LoginPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      attendant_name: "",
      password: "",
      message: ""
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.message) {
        this.setState({
          message: nextProps.message
        });
        nextProps.history.push("/");
      }
    else if (nextProps.error) {
      this.setState({
        message: nextProps.error
      });
    }
  }

  onChange(e) {
    e.preventDefault();
    const { target } = e;
    const { value } = target;
    const { name } = target;
    this.setState({
      [name]: value
    });
  }

  onSubmit(e) {
    e.preventDefault();
    this.props.loginAction(this.state);
  }

  render() {
    return (
      <div>
        <Login
          onChange={this.onChange}
          onSubmit={this.onSubmit}
          message={this.state.message}
          loading={this.state.loading}
        />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  successState: state.authReducer.isSuccessful,
  error: state.authReducer.error,
  message: state.authReducer.message,
  token: state.authReducer.token
});

export default connect(
  mapStateToProps,
  { loginAction }
)(LoginPage);
