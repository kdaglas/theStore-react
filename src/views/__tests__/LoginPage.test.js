import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage, mapStateToProps } from '../LoginPage';


describe('LoginPage', () => {
  it('should render without crashing', () => {
    const props = {};
    const instance = new LoginPage(props);
    const wrapper = shallow(instance.render());
    expect(wrapper).toMatchSnapshot();
  });

  it('should redirect on acquiring a token', () => {
    const push = jest.fn()
    const props = {
      history: {push},
    };
    const wrapper = shallow(<LoginPage {...props} />);

    wrapper.setProps({
      token: '123',
      user: 'wwwww',
      errors: '',
      isSuccessful: true,
      message:'message'
    });
    expect(props.history.push).toHaveBeenCalled();
  });

  it('should not reload if there is no token received', () => {
    const props = {
      token: '',
    };
    global.setTimeout = jest.fn();

    const wrapper = shallow(<LoginPage {...props} />);
    wrapper.instance().componentWillReceiveProps({});
    expect(global.setTimeout).not.toHaveBeenCalled();
  });

//   it('should call the login action', () => {
//     const props = { loginAction: jest.fn() };
//     const wrapper = shallow(<LoginView {...props} />);

//     const event = {
//       preventDefault: jest.fn(),
//       target: {
//         elements: {
//           email: {
//             value: 'testemail@mail.com',
//           },
//           password: {
//             value: 'Password1',
//           },
//         },
//       },
//     };
//     wrapper.instance().handleSubmit(event);
//     expect(wrapper.instance().props.loginAction).toHaveBeenCalled();
//   });
});