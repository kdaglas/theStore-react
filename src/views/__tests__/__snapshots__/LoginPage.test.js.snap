// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should render without crashing 1`] = `ShallowWrapper {}`;
